PcsConfig/DeviceGroupsTests.cs

```diff
@@ -0,0 +1,280 @@
+    {
+        private readonly IHttpClient httpClient;
+        private const int TIMEOUT = 10000;
+        private List<string> disposeList = new List<string>();
+
+        public DeviceGroupsTests()
+        {
+            httpClient = new HttpClient();
+        }
+
+        /// <summary>
+        /// Test that the service can create a new Device Group
+        /// </summary>
+        [Fact, Trait(Constants.TEST, Constants.INTEGRATION_TEST)]
+        public void ShouldCreateNewDeviceGroup()
+        {
+            // Arange
+            var inputDeviceGroup = new DeviceGroupApiModel
+            {
+                Id = "",
+                DisplayName = "TEST-CreateDeviceGroup",
+                Conditions = CreateFakeDeviceGroupConditionList(),
+                ETag = "",
+                Metadata = null
+            };
+            var request = new HttpRequest();
+            request.SetContent(inputDeviceGroup);
+            request.SetUriFromString(Constants.CONFIG_ADDRESS + "/devicegroups");
+            request.Options.Timeout = TIMEOUT;
+
+            // Act
+            var response = this.httpClient.PostAsync(request).Result;
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var createdDeviceGroup = JsonConvert.DeserializeObject<DeviceGroupApiModel>(response.Content);
+            Assert.Equal(inputDeviceGroup.DisplayName, createdDeviceGroup.DisplayName);
+            Assert.Equal(inputDeviceGroup.Conditions.ToString(), createdDeviceGroup.Conditions.ToString());
+
+            // Dispose
+            disposeList.Add(createdDeviceGroup.Id);
+        }
+
+        /// <summary>
+        /// Test that the service can update a created Device Group
+        /// </summary>
+        [Fact, Trait(Constants.TEST, Constants.INTEGRATION_TEST)]
+        public void ShouldUpdateCreatedDeviceGroup()
+        {
+            // Arange for create
+            var inputDeviceGroup = new DeviceGroupApiModel
+            {
+                Id = "",
+                DisplayName = "TEST-CreateDeviceGroup",
+                Conditions = CreateFakeDeviceGroupConditionList(),
+                ETag = "",
+                Metadata = null
+            };
+            var request = new HttpRequest();
+            request.SetContent(inputDeviceGroup);
+            request.SetUriFromString(Constants.CONFIG_ADDRESS + "/devicegroups");
+            request.Options.Timeout = TIMEOUT;
+
+            // Act for create
+            var response = this.httpClient.PostAsync(request).Result;
+
+            // Assert for create
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var createdDeviceGroup = JsonConvert.DeserializeObject<DeviceGroupApiModel>(response.Content);
+            Assert.Equal(inputDeviceGroup.DisplayName, createdDeviceGroup.DisplayName);
+            Assert.Equal(inputDeviceGroup.Conditions.ToString(), createdDeviceGroup.Conditions.ToString());
```

Comment:
Move the the Create to a private method and call it from `ShouldCreateNewDeviceGroup` and `ShouldUpdateCreatedDeviceGroup`

Reply:
done